community-tutorials/how-to-install-postgres-and-pgadmin-with-docker-and-connect-pgmadmin-with-postgres/01-en.md

```diff
@@ -0,0 +1,156 @@
+---
+title: How to install PostgreSQL and pgAdmin with docker and connect pgAdmin with PostgreSQL
+description: This tutorial describes how to install a PostgreSQL database and pgAdmin. pgAdmin is a web based administration tool for the PostgreSQL database. 
+updated_at: 2021-11-07
+slug: how-to-install-postgres-and-pgadmin-with-docker-and-connect-pgadmin-with-postgres
+author_name: Patrick Weber
+author_url: -
+author_image: -
+author_bio: -
+tags: debian, ubuntu, docker, pgAdmin, PostgreSQL
+netcup_product_url: https://www.netcup.de/vserver/vps.php
+language: en
+available_languages: en
+---
+
+# Introduction
+This tutorial is about how to install PostgreSQL and pgAdmin with docker and connect pgAdmin with PostgreSQL.
+
+# Requirements
+First you need a virtual server with debian or ubuntu from netcup. This tutorial requires also the setup of docker and docker-compose. On the netcup-community-tutorials you find another tutorial how to setup docker and docker-compose on ubuntu.
+
+# Step 1 - First check your docker and docker-compose version
+Check the docker version: ```docker -v```
+Output should look like this, maybe with another version: ```Docker version 20.10.10, build b485636```
+
+Check the docker-compose version: ```docker-compose version```
+Output should look like this, maybe with another version: ```Docker Compose version v2.1.0```
+
+# Step 4 - Create a compose.yaml file, with PostgreSQL and pgAdmin
+Create the file compose.yaml file with nano: ```nano ./compose.yaml```
+
+```
+services:
+  db:
+    image: postgres:14.0
+    container_name: db
+    restart: always
+    ports:
+      - 5432:5432
+    environment:
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_USER=postgres
+    volumes:
+      - db_volume:/var/lib/postgresql
+    networks:
+      backend_network:
+  pgadmin:
+    image: dpage/pgadmin4:6.1
+    container_name: pgadmin
+    restart: always
+    environment:
+      - PGADMIN_DEFAULT_EMAIL=admin@your-netcup-domain.net
+      - PGADMIN_DEFAULT_PASSWORD=secret
+      - PGADMIN_LISTEN_PORT=8080
+    ports:
+      - "8080:8080"
+    volumes:
+      - pgadmin_volume:/var/lib/pgadmin
+    networks:
+      backend_network:
+volumes:
+  db_volume:
+  pgadmin_volume:
+
+networks:
+  backend_network:
+```
+
+Save the compose.yaml file.
+
+Here are two services defined: One for PostgreSQL and one for pgAdmin. Both services have a fix container-name and both have its own volume. In the environments-definition you can see the credentials to access the services. The postgres servuce will be exposed on port 5432 to the outside. pgadmin will be exposed on port 8080 to the outside. Both services are using the same docker network "backend_network", which is also defined in the networks-definition.
+
+# Step 5 - Start all docker services
+To start all docker services run: ```docker-compose up -d```
```

Comment:
ERROR: 
        Can't find a suitable configuration file in this directory or any
        parent. Are you in the right directory?
        Supported filenames: **docker-compose.yml, docker-compose.yaml**

Reply:
if i rename the file like "docker-compose.yaml"
i get another error:
```
Unsupported config option for networks: 'backend_network'
Unsupported config option for services: 'db'
Unsupported config option for volumes: 'db_volume'
```

Reply:
i moved the file from compose.yaml to docker-compose.yaml.
I also added the version: '3' in the docker-compose-file, to support older versions.

Can you pull the latest changes and try the docker-compose file again?